torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py

```diff
@@ -190,7 +190,7 @@ def __init__(
         rank: int,
     ):
         self.rank = rank
-        self.trainer_group = dist.new_group(TRAINER_RANKS)
+        self.trainer_group = dist.new_group(TRAINER_RANKS) if ddp_mode in (DdpMode.INSIDE, DdpMode.OUTSIDE) else None
```

Comment:
Why the old code was flaky? Is it because creating without using a PG could lead to race in destruction?

Reply:
I wasn't able to repro the exact failure, so I'm not sure of the exact problem here. Looking at the code it seems all processes execute `init_process_group` before we reach here, so I'm not sure what might've led to the original failure.